ApiFirst/ApiFirst/Controllers/CostumersController.cs

```diff
@@ -0,0 +1,89 @@
+﻿using DomainModels.Models;
+using DomainServices.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+
+namespace WebApi.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CustomersController : ControllerBase
+    {
+        private readonly ICustomerRepository _service;
+
+        public CustomersController(ICustomerRepository repo)
+        {
+            _service = repo ?? throw new ArgumentNullException(nameof(repo));
+        }
+
+        [HttpGet("get-by-cpf")]
+        public Customer? GetByCpf([FromQuery] string cpf)
+        {
+            return _service.GetByCpf(cpf);
+        }
+
+
+        [HttpGet("get-all")]
```

Comment:
É redundante ter get na rota porque o endpoint já é get
```suggestion
        [HttpGet]
```